[BUG] `Schema.oneOfWrapped` does not compile in Scala 3 for both the `enum`s and `sealed trait`s

Tapir version: 1.6.0

Scala version: 3.3.0

**Describe the bug**

What is the problem?

Schema derivation for enums and sealed traits fails to compile in Scala 3.

**How to reproduce?**

Compiling the following code:
```scala
//> using scala "3.3.0"
//> using dep "com.softwaremill.sttp.tapir::tapir-json-circe:1.6.0"

import sttp.tapir.*
import sttp.tapir.json.circe.*
import io.circe.Codec

enum Fruit derives Codec.AsObject:
  case Apple, Banana

given Schema[Fruit] = Schema.oneOfWrapped

val fruits = endpoint.post.in("fruits").in(jsonBody[Fruit])

@main def main(): Unit = println("")
```

fails with the error:
```scala
Main.scala:12:23: Exception occurred while executing macro expansion.
java.lang.AssertionError: assertion failed
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:11)
	at scala.quoted.runtime.impl.QuotesImpl$reflect$TypeIdent$.apply(QuotesImpl.scala:1102)
	at scala.quoted.runtime.impl.QuotesImpl$reflect$TypeIdent$.apply(QuotesImpl.scala:1101)
	at sttp.tapir.macros.SchemaCompanionMacros$.$anonfun$2(SchemaMacros.scala:254)
	at scala.collection.immutable.List.map(List.scala:246)
	at sttp.tapir.macros.SchemaCompanionMacros$.generateOneOfWrapped(SchemaMacros.scala:264)

 [12:23]
```

**Additional information**
Substituting 
```scala
enum Fruit derives Codec.AsObject:
  case Apple, Banana
```
 with 
```scala
sealed trait Fruit
object Fruit:
  case object Apple extends Fruit
  case object Banana extends Fruit
given Codec[Fruit] = Codec.AsObject.derived
```
yields the same result.

--- a/core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala
+++ b/core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala
@@ -248,20 +248,23 @@ private[tapir] object SchemaCompanionMacros {
     if (!symbol.isClassDef || !(symbol.flags is Flags.Sealed)) {
       report.errorAndAbort("Can only generate a coproduct schema for an enum, sealed trait or class.")
     } else {
-      val subclasses = symbol.children.toList.sortBy(_.name)
+      val children = symbol.children.toList.sortBy(_.name)
 
-      val subclassesSchemas: List[Expr[(String, Schema[_])]] = subclasses.map(subclass =>
-        TypeIdent(subclass).tpe.asType match {
-          case '[f] => {
-            Expr.summon[Schema[f]] match {
-              case Some(subSchema) => '{ ${ Expr(subclass.name) } -> Schema.wrapWithSingleFieldProduct(${ subSchema })($conf) }
-              case None => {
-                val typeName = TypeRepr.of[f].typeSymbol.name
-                report.errorAndAbort(s"Cannot summon schema for `${typeName}`. Make sure schema derivation is properly configured.")
+      val childSchemas: List[Expr[(String, Schema[_])]] = children.map(child =>
+        if child.isClassDef
+        then // this can be a type (enum case with params / case class with params), or a parameterless enum case / case object
+          TypeIdent(child).tpe.asType match {
+            case '[f] => {
+              Expr.summon[Schema[f]] match {
+                case Some(subSchema) => '{ ${ Expr(child.name) } -> Schema.wrapWithSingleFieldProduct(${ subSchema })($conf) }
+                case None => {
+                  val typeName = TypeRepr.of[f].typeSymbol.name
+                  report.errorAndAbort(s"Cannot summon schema for `${typeName}`. Make sure schema derivation is properly configured.")
+                }
               }
             }
           }
-        }
+        else '{ ${ Expr(child.name) } -> Schema(SchemaType.SProduct[E](Nil), name = Some(Schema.SName(${ Expr(child.name) }))) }
       )
 
       def subtypeSchema(e: Expr[E], map: Expr[Map[String, Schema[_]]]) = {
@@ -272,12 +275,10 @@ private[tapir] object SchemaCompanionMacros {
 
         val t = Match(
           eIdent,
-          subclasses.map { subclass =>
-            CaseDef(
-              Typed(Wildcard(), TypeIdent(subclass)),
-              None,
-              Block(Nil, '{ Some(SchemaWithValue($map(${ Expr(subclass.name) }).asInstanceOf[Schema[Any]], $e)) }.asTerm)
-            )
+          children.map { child =>
+            val caseThen = Block(Nil, '{ Some(SchemaWithValue($map(${ Expr(child.name) }).asInstanceOf[Schema[Any]], $e)) }.asTerm)
+            if child.isClassDef then CaseDef(Typed(Wildcard(), TypeIdent(child)), None, caseThen)
+            else CaseDef(Ident(child.termRef), None, caseThen)
           }
         )
 
@@ -291,7 +292,7 @@ private[tapir] object SchemaCompanionMacros {
         import _root_.sttp.tapir.SchemaType._
         import _root_.scala.collection.immutable.{List, Map}
 
-        val subclassNameToSchema: List[(String, Schema[_])] = List(${ Varargs(subclassesSchemas) }: _*)
+        val subclassNameToSchema: List[(String, Schema[_])] = List(${ Varargs(childSchemas) }: _*)
         val subclassNameToSchemaMap: Map[String, Schema[_]] = subclassNameToSchema.toMap
 
         val sname = SName(SNameMacros.typeFullName[E], ${ Expr(typeParams) })

--- a/core/src/test/scala-3/sttp/tapir/SchemaMacroScala3Test.scala
+++ b/core/src/test/scala-3/sttp/tapir/SchemaMacroScala3Test.scala
@@ -1,0 +1,23 @@
+package sttp.tapir
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.duration.Duration
+
+class SchemaMacroScala3Test extends AnyFlatSpec with Matchers:
+  import SchemaMacroScala3Test._
+
+  it should "validate enum" in {
+    given s: Schema[Fruit] = Schema.oneOfWrapped
+
+    s.schemaType should matchPattern { case SchemaType.SCoproduct(_, _) => }
+
+    val coproduct = s.schemaType.asInstanceOf[SchemaType.SCoproduct[Fruit]]
+    coproduct.subtypeSchema(Fruit.Apple).map(_.schema) shouldBe Some(Schema(SchemaType.SProduct(Nil), Some(Schema.SName("Apple"))))
+    coproduct.subtypeSchema(Fruit.Apple).map(_.value) shouldBe Some(Fruit.Apple)
+  }
+
+object SchemaMacroScala3Test:
+  enum Fruit:
+    case Apple, Banana

